Dockerfile

```diff
@@ -0,0 +1,397 @@
+    cd /usr/local/bin && \
+    chmod a+rx ccx
+
+# VTK v8.2.0
+RUN git clone -n https://gitlab.kitware.com/vtk/vtk.git && \
+    mkdir /tmp/vtk/build && cd /tmp/vtk/build && \
+    git checkout v8.2.0 && \
+    cmake -DVTK_QT_VERSION:STRING=5 \
+    -D QT_QMAKE_EXECUTABLE:PATH=/usr/local/Qt-5.12.6/bin/qmake \
+    -D VTK_Group_Qt:BOOL=ON \
+    -D CMAKE_PREFIX_PATH:PATH=/usr/local/Qt-5/lib/cmake  \
+    -D BUILD_SHARED_LIBS:BOOL=ON \
+    .. && \
+    make -j $(nproc --ignore=2) && \
+    make -j $(nproc --ignore=2) install && \
+    rm -rfv /tmp/*
+
+# HDF5 v1.10.6 (CMake version)
+ENV hdf5_path=/usr/local/hdf5
+RUN wget https://hdf-wordpress-1.s3.amazonaws.com/wp-content/uploads/manual/HDF5/HDF5_1_10_6/source/CMake-hdf5-1.10.6.tar.gz && \
+    tar -xzf CMake-hdf5-1.10.6.tar.gz && \
+    cd CMake-hdf5-1.10.6 && \
+    ./build-unix.sh && \
+    ./HDF5-1.10.6-Linux.sh --exclude-subdir --skip-license && \
+    mkdir $hdf5_path && \
+    mv HDF_Group/HDF5/1.10.6/* $hdf5_path && \
+    export PATH=$PATH:$hdf5_path && \
+    echo "$hdf5_path/lib" > /etc/ld.so.conf.d/hdf5.conf && \
+    ldconfig && \
+    # Because HDF5's libz.so does not have version information, link to use
+    # package installed version.
+    rm /usr/local/hdf5/lib/libz.so* && \
+    ln -s /usr/lib/x86_64-linux-gnu/libz.so /usr/local/hdf5/lib/libz.so && \
+    ln -s /usr/lib/x86_64-linux-gnu/libz.so /usr/local/hdf5/lib/libz.so.1 && \
+    ln -s /usr/lib/x86_64-linux-gnu/libz.so /usr/local/hdf5/lib/libz.so.1.2.11 && \
+    rm -rfv /tmp/* 
+
+# HDF5's root has to be on the path for it to be detected.
+ENV PATH=$PATH:$hdf5_path
+
+# Libmed v3.0.6-11 (AKA: MED-fichier/Modelisation and Data Exchange)
+RUN wget https://salsa.debian.org/science-team/med-fichier/-/archive/debian/4.0.0+repack-9/med-fichier-debian-4.0.0+repack-9.tar.gz && \
+    tar -xzf med-fichier-debian-4.0.0+repack-9.tar.gz && \
+    rm med-fichier-debian-4.0.0+repack-9.tar.gz && \
+    mkdir /tmp/med-fichier-debian-4.0.0+repack-9/build && \
+    cd /tmp/med-fichier-debian-4.0.0+repack-9/build && \
+    cmake .. && \
+    make -j $(nproc --ignore=2) && make -j $(nproc --ignore=2) install && \
+    rm -rfv /tmp/*
+
+# OpenCamLib v29/12/2019(Commit:983a4168fb0a8e84154c45c6f0a286dc2e752b9a)
+RUN git clone -n https://github.com/aewallin/opencamlib.git && \
+    mkdir /tmp/opencamlib/build && cd /tmp/opencamlib/build && \
+    git checkout 983a4168fb0a8e84154c45c6f0a286dc2e752b9a && \
+    cmake -D BUILD_PY_LIB=ON -D BUILD_CXX_LIB=OFF -D USE_PY_3=ON ../src && \
+    make -j $(nproc --ignore=2) && \
+    make -j $(nproc --ignore=2) install
+
+# Add the build script
+ADD add_files/freecad_build_script.sh /root/build_script.sh
+
+# Add enviroment varaible so CMake can find QT5
+ENV CMAKE_PREFIX_PATH=/usr/local/Qt-5
+
+# So Qt5 can find it's shared libaries
+RUN echo "/usr/local/Qt-5/lib" > /etc/ld.so.conf.d/qt5.conf && \
+    ldconfig
+
+WORKDIR /root
+
+# Note for later: May need -fPIC
```

Comment:
What may need `-fPIC`?
AFAIK it's never bad to use `-fPIC` (unless maybe you are doing lowlevel embedded stuff)